### PR TITLE
worktracker: enforce random_interval after one failure

### PR DESCRIPTION
**Pull request by vepkenez**

### CODE REVIEW

File: nucypher/blockchain/eth/token.py [hunk 1]
```diff
@@ -750,6 +759,7 @@ def _do_work(self) -> None:
             return  # This cycle is finished.
         else:
             # Randomize the next task interval over time, within bounds.
+            self._consecutive_fails = 0
```
Comment (vepkenez) [inline nucypher/blockchain/eth/token.py:764]: The logic flow of this method is pretty hard to follow.  I _think_ this is the spot where we can consider this period's work "successful" and reset this value.

Comment (cygnusv) [reply nucypher/blockchain/eth/token.py:764]: This block is reached when there are no pending TXs but we haven't confirmed next period yet. Perhaps this is not the place you were thinking. 

I think the following two places are better suited to reset that variable:
* Line 748: This is the block where we know that the next period is committed. 
* Line 737: In this block, we detect that the period has changed, so it doesn't matter if the worker was successful or not in the previous period.
